hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockTest.groovy

```diff
@@ -276,11 +283,73 @@ class HermesMockTest extends Specification {
             1 == hermes.query().countMatchingJsonMessages(topicName, TestMessage, filter)
     }
 
+    def "should reset received requests with Avro messages"() {
+        given:
+        def topicName = "my-test-avro-topic"
+        hermes.define().avroTopic(topicName)
+        publish(topicName, new TestMessage("test-key", "test-value"))
+        assert hermes.query().countAvroMessages(topicName) == 1
+
+        when:
+        hermes.resetReceivedAvroRequests(topicName, schema, TestMessage, {it -> it.key == "test-key"})
+
+        then:
+        hermes.query().countAvroMessages(topicName) == 0
+    }
+
+    def "should not reset received requests with Avro messages when predicate does not match"() {
+        given:
+        def topicName = "my-test-avro-topic"
+        hermes.define().avroTopic(topicName)
+        publish(topicName, new TestMessage("test-key", "test-value"))
+
+        when:
+        hermes.resetReceivedAvroRequests(topicName, schema, TestMessage, {it -> it.key == "different-test-key"})
+
+        then:
+        hermes.query().countAvroMessages(topicName) == 1
+    }
+
+    def "should reset received requests with JSON messages"() {
+        given:
+        def topicName = "my-test-json-topic"
+        hermes.define().avroTopic(topicName)
+        publish(topicName, new TestMessage("test-key", "test-value").asJson())
+        hermes.query().countJsonMessages(topicName, TestMessage) == 1
+
+        when:
+        hermes.resetReceivedJsonRequests(topicName, TestMessage, {it -> it.key == "test-key"})
+
+        then:
+        hermes.query().countJsonMessages(topicName, TestMessage) == 0
+    }
+
+    def "should not reset received requests with JSON messages when predicate does not match"() {
+        given:
+        def topicName = "my-test-json-topic"
+        hermes.define().avroTopic(topicName)
+        publish(topicName, new TestMessage("test-key", "test-value").asJson())
+
+        when:
+        hermes.resetReceivedJsonRequests(topicName, TestMessage, {it -> it.key == "different-test-key"})
+
+        then:
+        hermes.query().countJsonMessages(topicName, TestMessage) == 1
+    }
+
     def publish(String topic) {
         publish(topic, TestMessage.random().asJson())
     }
 
     def publish(String topic, String body) {
         publisher.publish(topic, body)
     }
+
+    def publish(String topic, TestMessage message) {
```

Comment:
And this function to `publishAvro`, to make it clear what is send :)

Reply:
Good idea. Done.